ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/computation/CommandRunnerTest.java

```diff
@@ -53,29 +70,131 @@ public void shouldFetchAndRunNewCommandsFromCommandTopic() {
           expectLastCall();
         }
     );
-    replay(statementExecutor);
+    expect(commandStore.getNewCommands()).andReturn(commands);
+    replay(statementExecutor, commandStore);
+    final CommandRunner commandRunner = new CommandRunner(statementExecutor, commandStore, 1);
+
+    // When:
+    commandRunner.fetchAndRunCommands();
+
+    // Then:
+    verify(statementExecutor);
+  }
 
-    final CommandStore commandStore = mock(CommandStore.class);
+  @Test
+  public void shouldRetryCommands() {
+    // Given:
+    final List<QueuedCommand> commands = Collections.singletonList(getQueuedCommands().get(0));
+    final QueuedCommand command = commands.get(0);
+    statementExecutor.handleStatement(
+        same(command.getCommand().get()),
+        same(command.getCommandId()),
+        same(command.getStatus()));
+    expectLastCall().andThrow(new RuntimeException("Something bad happened"));
```

Comment:
Does EasyMock not support chaining? e.g.

```
expectLastCall()
    .anyTimes()
    .andThrow(e) .       <- first invocation result
    .andAnswer(blah)  <- second invocation result
```

Mockito does... 

(IMHO Mockito is clearly superior to EasyMock. I keep coming up against things with EasyMock that I can't mock, which I would be able to do easily in Mockito)

Reply:
Not sure - I'll try it.

Reply:
I'm going to punt this and the other CommandRunnerTest comment to another PR I'm working on, which changes this stuff quite a bit.